src/order/complete_lattice.lean

```diff
@@ -568,11 +576,21 @@ le_antisymm
   (le_infi $ assume i, le_infi $ assume j, infi_le _ _)
   (le_infi $ assume ⟨i, h⟩, infi_le_of_le i $ infi_le _ _)
 
+/-- The symmetric case of `infi_and`, useful for rewriting into a infimum over a conjunction -/
```

Comment:
What happens if you try to use `infi_and` directly? And do you understand what some lemmas only half use the notation?

Reply:
If you try to prove `infi_and'` with `rw ← infi_and` you get:
```
rewrite tactic failed, did not find instance of the pattern in the target expression
  ⨅ (h₁ : ?m_2) (h₂ : ?m_4), ?m_5 ⟨h₁, h₂⟩
```
The thing it needs to rewrite is
```
(⨅ (h₁ : p) (h₂ : q), s h₁ h₂)
```
rewrite is not smart enough to realize that if `?m_5` is `fun x => s x.1 x.2` then it does unify.

I think some lemmas write `infi s` or `supr s` because of laziness/shorter to write that.